File: benchmarks/tsb/bench_series_cummax.ts
/**
 * Benchmark: series cummax on 100k-element Series
 */
import { Series, cummax } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data = Float64Array.from({ length: ROWS }, (_, i) => Math.sin(i * 0.01));
const s = new Series(data);

for (let i = 0; i < WARMUP; i++) {
  cummax(s);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  cummax(s);
}
const total = performance.now() - start;

console.log(JSON.stringify({ function: "series_cummax", mean_ms: total / ITERATIONS, iterations: ITERATIONS, total_ms: total }));

File: benchmarks/pandas/bench_series_cummax.py
"""Benchmark: series cummax on 100k-element Series"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = np.sin(np.arange(ROWS) * 0.01)
s = pd.Series(data)

for _ in range(WARMUP):
    s.cummax()

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.cummax()
total = (time.perf_counter() - start) * 1000

print(json.dumps({ "function": "series_cummax", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total }))
